doc build (sphinx) is broken on Ubuntu 20.04 (focal)

### What happened?

Between 3.3.0 and 3.3.2, the doc build failed on the apt.postgresql.org build infrastructure for Ubuntu focal like this (see e.g. https://pgdgbuild.dus.dg-i.net/job/patroni-binaries/architecture=amd64,distribution=focal/246/console):

```
21:37:19 Creating file /<<PKGBUILDDIR>>/docs/modules/modules.rst.
21:37:19 The builder is: html
21:37:19 building [mo]: targets for 0 po files that are out of date
21:37:19 building [html]: targets for 25 source files that are out of date
21:37:19 updating environment: 
21:37:19 Exception occurred:
21:37:19   File "/<<PKGBUILDDIR>>/docs/conf.py", line 302, in env_get_outdated
21:37:19     for doc in env.found_docs:
21:37:19 AttributeError: 'StandaloneHTMLBuilder' object has no attribute 'found_docs'
21:37:19 The full traceback has been saved in /tmp/sphinx-err-t8w33jek.log, if you want to report the issue to the developers.
21:37:19 Please also report this if it was a user error, so that a better error message can be provided next time.
21:37:19 A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
21:37:19 make[2]: *** [Makefile:20: html] Error 2
21:37:19 make[2]: Leaving directory '/<<PKGBUILDDIR>>/docs'
21:37:19 make[1]: *** [debian/rules:17: override_dh_auto_build] Error 2
21:37:19 make[1]: Leaving directory '/<<PKGBUILDDIR>>'
21:37:19 make: *** [debian/rules:8: binary] Error 2
```

I have not bisected this yet as I don't have a local focal build environment. Ubuntu 20.04 is currently still supported on apt.postgresql.org, but if this is too demanding to fix, we can drop focal packages for patroni.

### How can we reproduce it (as minimally and precisely as possible)?

Build sphinx documentation on Ubuntu focal

### What did you expect to happen?

build succeeds

### Patroni/PostgreSQL/DCS version

-   Patroni version: 3.3.2
-   PostgreSQL version: n/a
-   DCS (and its version): n/a


### Patroni configuration file

```yaml
n/a
```


### patronictl show-config

```yaml
n/a
```


### Patroni log files

```shell
n/a
```


### PostgreSQL log files

```shell
n/a
```


### Have you tried to use GitHub issue search?

- [X] Yes

### Anything else we need to know?

_No response_

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -289,13 +289,15 @@ def env_get_outdated(app, env, added, changed, removed):
     Remove the items listed in `docs_to_remove` from known pages.
     """
     to_remove = set()
-    for doc in env.found_docs:
-        if _to_be_removed(doc):
-            to_remove.add(doc)
+    if hasattr(env, 'found_docs'):
+        for doc in env.found_docs:
+            if _to_be_removed(doc):
+                to_remove.add(doc)
     added.difference_update(to_remove)
     changed.difference_update(to_remove)
     removed.update(to_remove)
-    env.project.docnames.difference_update(to_remove)
+    if hasattr(env, 'project'):
+        env.project.docnames.difference_update(to_remove)
     return []
 
 

